### ISSUE TITLE
Scroll paint dpi unstable

### ISSUE THREAD

**Issue by AaronVanGeffen** [OpenRCT2/OpenRCT2#11238]
**OS:** Arch Linux
**Version:** 0.2.5+
**Commit/Build:** 12ab701a0fc89e1e12a3724ebef164ae4c111172

I've encountered some strange behaviour in the `scroll_paint` event while working on #11231. Concretely, in window_shortcut_scrollpaint I'm finding the `dpi->width` to fluctuate massively. Is this expected behaviour? This is with multithreading off, for what it's worth.

Alternatively, I tried to use the widgets to compute the width:
```
const auto& widget = w->widgets[WWT_SCROLL];
const auto width = widget.right - widget.left;
```
But I find both .right and .left are set to 0. Expected would be 415 and 4, respectively.

For #11231, I've worked around this by using `w->width - 10` to get the right value, but I still think there is a bug or two to fix here.

**Comment by duncanspumpkin** [comment]
Yes DPI width can change depending on how the window is cut up for dirty drawing.
